googleapiclient/discovery.py

```diff
@@ -253,8 +262,13 @@ def build_from_document(
   # future is no longer used.
   future = {}
 
-  if isinstance(service, basestring):
-    service = json.loads(service)
+  try:
+    if isinstance(service, basestring):
+      service = json.loads(service)
+  except NameError:
+    if isinstance(service, str):
+      service = json.loads(service)
```

Comment:
I don't like this try~except clause.
How about adding this code on top of this module?

```python
try:
    basestring
except NameError:
    basestring = str
```